pseudopotential_calculator/calculations/slab.py

```diff
@@ -119,11 +126,56 @@ def plot_energy_against_n_vacuum_layer(
         atom = get_calculator_atom(calculator)
 
         energies.append(get_atom_potential_energy(atom))
-        n_vacuum_layers = get_n_vacuum_layers_from_out_put(atom)
+        n_vacuum_layers = _get_n_vacuum_layers_from_slab(atom)
         n_vacuum_layer.append(n_vacuum_layers)
 
     return plot_data_comparison(
         ("Vacuum Height", np.array(n_vacuum_layer), "N layers"),
         ("Energy", np.array(energies), "J"),
         ax=ax,
     )
+
+
+def get_atom_displacement(atom: Atoms, n_th_layer: int) -> float:
+    heights = cast(list[float], atom.positions[:, 2])  # type: ignore bad lib
+    n_cu_layer = len(np.unique(heights))
+    sorted_heights = np.unique(heights)[::-1]
+    atom_height = sorted_heights[n_th_layer - 1]
+    height_per_fixed_layer = sorted_heights[-2]
```

Comment:
split this height logic into another function and share with the vacuum convergence plot